packages/core/src/machines/actors/auth/signUp.ts

```diff
@@ -0,0 +1,229 @@
+import { createMachine, assign, sendUpdate } from 'xstate';
+import { passwordMatches, runValidators } from '../../../validators';
+
+import { AuthEvent, SignUpContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const signUpActor = createMachine<SignUpContext, AuthEvent>(
+  {
+    id: 'signUpActor',
+    initial: 'init',
+    context: {
+      remoteError: '',
+      formValues: {},
+      validationError: {},
+    },
+    states: {
+      init: {
+        always: [
+          { target: 'confirmSignUp', cond: 'shouldInitConfirmSignUp' },
+          { target: 'signUp' },
+        ],
+      },
+      signUp: {
+        type: 'parallel',
+        exit: ['clearError', 'clearFormValues'],
+        states: {
+          validation: {
+            initial: 'pending',
+            states: {
+              pending: {
+                invoke: {
+                  src: 'validateFields',
+                  onDone: {
+                    target: 'valid',
+                    actions: 'clearValidationError',
+                  },
+                  onError: {
+                    target: 'invalid',
+                    actions: 'setFieldErrors',
+                  },
+                },
+              },
+              valid: { entry: sendUpdate() },
+              invalid: { entry: sendUpdate() },
+            },
+            on: {
+              CHANGE: {
+                actions: 'handleInput',
+                target: '.pending',
+              },
+            },
+          },
+          submission: {
+            initial: 'idle',
+            states: {
+              idle: {
+                entry: sendUpdate(),
+                on: {
+                  SUBMIT: 'validate',
+                  FEDERATED_SIGN_IN: 'federatedSignIn',
+                },
+              },
+              federatedSignIn: {
```

Comment:
This is probably better named `federatedSignUp`... but leaving this as now to keep PR focused.